rllib/core/rl_module/rl_module.py

```diff
@@ -77,23 +100,106 @@ class RLModule(abc.ABC):
         More details here: https://github.com/pytorch/pytorch/issues/49726.
     """
 
-    @OverrideToImplementCustomLogic_CallToSuperRecommended
-    def __init__(self, config: Mapping[str, Any] = None) -> None:
-        self.config = config or {}
-        self.setup()
+    def __init_subclass__(cls, **kwargs):
+        # Automatically add a __post_init__ method to all subclasses of RLModule.
+        def init_decorator(previous_init):
+            def new_init(self, *args, **kwargs):
+                previous_init(self, *args, **kwargs)
+                if type(self) == cls:
+                    self.__post_init__()
+
+            return new_init
+
+        cls.__init__ = init_decorator(cls.__init__)
```

Comment:
you can't define this decorator somewhere, and just
```
@init_decorator
def __init__(self, ...):
    ...
```
?

Reply:
Then the sub-classes should also remember to add this decorator :) Which people may forget.

Reply:
:)